### ISSUE TITLE
OpenSSL Error: unsafe legacy renegotiation disabled

### ISSUE THREAD

**Issue by EndUser509** [mitmproxy/mitmproxy#5422]
#### Problem Description
I have switched from the CentOS AppStream Python 3.9 to a self-compiled Python 3.10.4. With `python.10` I sometimes get this error which I never had with 3.9 (I checked all my old logs):
```
Unable to establish TLS connection with server (OpenSSL Error([('SSL routines', '', 'unsafe legacy renegotiation disabled')])). Trying to establish TLS with client anyway.
```
So this error seems to occur in the server-facing connection where mitmproxy is the client. Renegotiation is AFAIK initiated by the client so it looks like some default has changed for the newer python.  
Before I start digging deeper myself I wonder if this rings a bell with someone.

#### Steps to reproduce the behavior:
1. Set up mitmproxy or mitmdump
2. Go to https://bild.de and click on a few links in a browser
3. 

#### System Information
Mitmproxy: 9.0.0.dev (+11, commit 6110a2a)
Python:    3.10.4
OpenSSL:   OpenSSL 3.0.3 3 May 2022
Platform:  Linux-4.18.0-383.el8.x86_64-x86_64-with-glibc2.28

**Comment by mhils** [comment]
This doesn't seem to be a mitmproxy bug, it looks like you are just dealing with a server that is so outdated that we cannot establish TLS securely.

**Comment by andrewpedia** [comment]
> For those having the same problem, here is my fix: [1ad96dd](https://github.com/mitmproxy/mitmproxy/commit/1ad96dd4b89beb3a71ffaabf959d889d060e7141)
> 
> I originally found the required flag here: https://stackoverflow.com/a/76045866/1584915
> 
> I wonder if this is worthwhile to make a PR, maybe this flag should be added when the flag `ssl_insecure` is passed. This issue is very annoying when the target server uses outdated SSL and is not under control.

how to do this on android ? thanks
